### PR TITLE
[#6688] [EIP-1577] Update ENS resolving implementation to match what is defined …

### LINKED ISSUE #6688: Update ENS resolving implementation to match what is defined in EIP-1577
**What problem are you trying to solve?**

To try and find a better system @arachnid and I created [EIP-1577](https://github.com/ethereum/EIPs/pull/1577) that allows for a better representation of network and content addresses.

**Describe the solution you'd like**

Implement resolution using the new `multiaddr` field when available.

*Issue was copied over from the [Metamask](https://github.com/MetaMask/metamask-extension/issues/5742) repository because of laziness*

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #6688

### CODE REVIEW

File: src/status_im/browser/core.cljs [hunk 1]
```diff
@@ -82,10 +82,17 @@
   (let [history-host (http/url-host (try (nth history history-index) (catch js/Error _)))]
     (assoc browser :unsafe? (dependencies/phishing-detect history-host))))
 
-(defn resolve-ens-multihash-callback [hex]
+(defn resolve-ens-content-callback [hex]
   (let [hash (when hex (multihash/base58 (multihash/create :sha2-256 (subs hex 2))))]
     (if (and hash (not= hash resolver/default-hash))
-      (re-frame/dispatch [:browser.callback/resolve-ens-multihash-success hash])
+      (re-frame/dispatch [:browser.callback/resolve-ens-multihash-success "01" hash])
```
Comment (StatusSceptre) [inline src/status_im/browser/core.cljs:88]: 01 ?

File: src/status_im/browser/core.cljs [hunk 2]
```diff
@@ -82,10 +82,17 @@
...
+      (re-frame/dispatch [:browser.callback/resolve-ens-contenthash]))))
+
+(defn resolve-ens-contenthash-callback [hex]
+  (let [proto-code (subs hex 2 4)
+        hash (when hex (multihash/base58 (multihash/create :sha2-256 (subs hex 8))))]
```
Comment (StatusSceptre) [inline src/status_im/browser/core.cljs:93]: Maybe extract this and have a small test?

File: src/status_im/browser/core.cljs [hunk 3]
```diff
@@ -97,13 +104,26 @@
...
+(fx/defn resolve-ens-contenthash
+  [{{:keys [web3 network] :as db} :db}]
+  (let [current-url (get-current-url (get-current-browser db))
+        host (http/url-host current-url)]
+    (let [network (get-in db [:account/account :networks network])
+          chain   (ethereum/network->chain-keyword network)]
+      {:db                            (update db :browser/options assoc :resolving? true)
+       :browser/resolve-ens-contenthash {:web3     web3
+                                         :registry (get ens/ens-registries
+                                                        chain)
```
Comment (StatusSceptre) [inline src/status_im/browser/core.cljs:123]: 2 lines?

**Review by StatusSceptre — COMMENTED** [review]
We don't have backwards support of old method with that right?

**Comment by jeluard** [comment]
@flexsurfer Do we maintain backwards compatibility ?

**Comment by flexsurfer** [comment]
@jeluard if you mean content field support, yes we do, so our dapps - ens names and others will still work without changes

File: src/status_im/browser/core.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/browser/core.cljs:88]: this is backwards support for content field, we supported only ipfs for it , https://github.com/ensdomains/multicodec/blob/master/table.csv#L3

Comment (jeluard) [reply src/status_im/browser/core.cljs:88]: Cool!

**Review by jeluard — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

**Comment by asemiankevich** [comment]
@flexsurfer please rebase to have iOS working too. jacob just merged the fix

**Comment by decanus** [comment]
@flexsurfer Just to be sure, did this implementation use the newest spec as found on: https://eips.ethereum.org/EIPS/eip-1577

<blockquote><div>Ethereum Improvement Proposals</div><div><strong><a href="http://eips.ethereum.org/EIPS/eip-1577">contenthash field for ENS</a></strong></div><div>Ethereum Improvement Proposals (EIPs) describe standards for the Ethereum platform, including core protocol specifications, client APIs, and contract standards.</div></blockquote>

**Comment by flexsurfer** [comment]
@decanus yes, it was implemented according to the latest spec

**Pull request merged**